app/src/main/java/woowacourse/movie/view/home/movies/adapter/MovieAdapter.kt

```
@@ -0,0 +1,57 @@
+        parent: ViewGroup,
+        viewType: Int,
+    ): RecyclerView.ViewHolder {
+        val inflater =
+            LayoutInflater.from(parent.context)
+
+        val movieItemBinding = MovieItemBinding.inflate(inflater, parent, false)
+        val advertisementItemBinding = AdvertisementItemBinding.inflate(inflater, parent, false)
+        return when (viewType) {
+            VIEW_TYPE_ADVERTISEMENT -> AdvertiseViewHolder(advertisementItemBinding)
+            VIEW_TYPE_MOVIE -> MovieViewHolder(movieItemBinding, onClickBooking)
+            else -> throw IllegalArgumentException()
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)